contracts/executable/AxelarGMPExecutableWithToken.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import { IAxelarGMPGatewayWithToken } from '../interfaces/IAxelarGMPGatewayWithToken.sol';
+import { IAxelarGMPExecutableWithToken } from '../interfaces/IAxelarGMPExecutableWithToken.sol';
+import { AxelarGMPExecutable } from './AxelarGMPExecutable.sol';
+
+/**
+ * @title AxelarGMPExecutableWithToken
+ * @dev Abstract contract to be inherited by contracts that need to execute cross-chain commands involving tokens via Axelar's Gateway.
+ * It extends AxelarGMPExecutable and implements the IAxelarGMPExecutableWithToken interface.
+ */
+abstract contract AxelarGMPExecutableWithToken is IAxelarGMPExecutableWithToken, AxelarGMPExecutable {
+    /// @dev Reference to the Axelar Gateway With Token contract.
+    IAxelarGMPGatewayWithToken public immutable gatewayWithToken;
```

Comment:
I think we should just declare the original one as an `address`, and cast it when needed in the logic, instead of adding two immutables.